scripts/l2systemDeploy.ts

```diff
@@ -0,0 +1,476 @@
+  );
+
+  await voltDAIPSM.deployed();
+
+  /// Deploy DAI Peg Stability Module
+  /// PSM will trade VOLT between 100 cents and 111 cents. If the Volt price exceeds 111 cents,
+  /// the floor price will have to be moved lower as the oracle price is inverted.
+  /// If price is outside of this band, the PSM will not allow trades
+  const voltUSDCPSM = await voltPSMFactory.deploy(
+    voltUSDCFloorPrice,
+    voltUSDCCeilingPrice,
+    {
+      coreAddress: core.address,
+      oracleAddress: oraclePassThrough.address, /// OPT
+      backupOracle: ethers.constants.AddressZero,
+      decimalsNormalizer: voltUSDCDecimalsNormalizer,
+      doInvert: true /// invert the price so that the Oracle and PSM works correctly
+    },
+    MINT_FEE_BASIS_POINTS,
+    L2_REDEEM_FEE_BASIS_POINTS,
+    reservesThreshold,
+    mintLimitPerSecond,
+    voltPSMBufferCap,
+    L2_ARBITRUM_USDC,
+    VOLT_FUSE_PCV_DEPOSIT /// intentionally set the PCV deposit as an address that does not exist on l2
+    /// any calls to try to allocate surplus will fail, however the reserves threshold is too high
+    /// so this call will never be be attempted in the first place
+  );
+
+  await voltUSDCPSM.deployed();
+
+  const PCVGuardian = await ethers.getContractFactory('PCVGuardian');
+  /// safe address is protocol multisig
+  const pcvGuardian = await PCVGuardian.deploy(core.address, L2_ARBITRUM_PROTOCOL_MULTISIG_ADDRESS, [
+    voltDAIPSM.address, /// whitelist psm's to withdraw from
+    voltUSDCPSM.address
+  ]);
+  await pcvGuardian.deployed();
+
+  const PCVGuardAdmin = await ethers.getContractFactory('PCVGuardAdmin');
+  const pcvGuardAdmin = await PCVGuardAdmin.deploy(core.address);
+  await pcvGuardAdmin.deployed();
+
+  /// -------- PCV Guardian Actions --------
+
+  // Grant PCV Controller and Guardian Roles to the PCV Guardian Contract
+  await core.grantPCVController(pcvGuardian.address);
+  await core.grantGuardian(pcvGuardian.address);
+
+  // Create the PCV_GUARD_ADMIN Role and Grant to the PCV Guard Admin Contract
+  await core.createRole(PCV_GUARD_ADMIN_ROLE, await core.GOVERN_ROLE());
+  await core.grantRole(PCV_GUARD_ADMIN_ROLE, pcvGuardAdmin.address);
+
+  // Create the PCV Guard Role and grant the role to PCV Guards via the PCV Guard Admin contract
+  await core.createRole(PCV_GUARD_ROLE, PCV_GUARD_ADMIN_ROLE);
+  await pcvGuardAdmin.grantPCVGuardRole(PCV_GUARD_EOA_1);
+  await pcvGuardAdmin.grantPCVGuardRole(PCV_GUARD_EOA_2);
+
+  await core.grantGovernor(L2_ARBITRUM_PROTOCOL_MULTISIG_ADDRESS); /// give multisig the governor role
+  await core.grantPCVController(L2_ARBITRUM_PROTOCOL_MULTISIG_ADDRESS); /// give multisig the PCV controller role
+
+  console.log('\n ~~~~~ Deployed Contracts Successfully ~~~~~ \n');
+
+  console.log(`⚡Core⚡:                 ${core.address}`);
+  console.log(`⚡VOLT⚡:                 ${volt}`);
+  console.log(`⚡PCVGuardian⚡:          ${pcvGuardian.address}`);
+  console.log(`⚡VOLT DAI PSM⚡:         ${voltDAIPSM.address}`);
```

Comment:
We should add a comment for the address for the USDC PSM here also

Reply:
Added